src/python/WMCore/Services/RucioConMon/RucioConMon.py

```diff
@@ -0,0 +1,110 @@
+#!/usr/bin/env python
+# coding=utf-8
+"""
+Rucio Consistency Monitor Service class developed on top of the native WMCore
+Service APIs providing custom output and handling as necessary for the
+CMS Workload Management system
+"""
+
+from __future__ import division, print_function, absolute_import
+
+from urllib.parse import urlencode
+
+import json
+import logging
+
+from WMCore.Services.Service import Service
+
+
+class RucioConMon(Service):
+    """
+    API for dealing with retrieving information from Rucio Consistency Monitor
+    """
+
+    def __init__(self, url, logger=None, configDict=None):
+        """
+        Init method for the RucioConMon Class
+        """
+        configDict = configDict or {}
+        configDict.setdefault('endpoint', url)
+        configDict.setdefault('cacheduration', 1)  # in hours
+        configDict.setdefault('accept_type', 'application/json')
+        configDict.setdefault('content_type', 'application/json')
+        configDict['logger'] = logger if logger else logging.getLogger()
+        super(RucioConMon, self).__init__(configDict)
+        self['logger'].debug("Initializing RucioConMon with url: %s", self['endpoint'])
+
+    def _getResult(self, uri, callname="", clearCache=True, args=None):
+        """
+        Either fetch data from the cache file or query the data-service
+        :param uri: The endpoint uri
+        :return:    A dictionary
+        """
+
+        # NOTE: Unlike the common case we are not using the callname for building
+        #       apiUrl. we  are using it only for giving proper name of the cache
+        #       file.
+        # NOTE: The 'callname' should not contain '/', otherwise the cache is
+        #       tried to be created with a a non existing subdirectory structure
+
+        cachedApi = "%s.json" % callname
+        # apiUrl = '%s?json&preset=%s' % (uri, callname)
+        apiUrl = uri
+
+        self['logger'].debug('Fetching data from %s, with args %s', apiUrl, args)
+        if args:
+            apiUrl = "%s&%s" % (apiUrl, urlencode(args, doseq=True))
+
+        if clearCache:
+            self.clearCache(cachedApi, args)
+        data = self.refreshCache(cachedApi, apiUrl)
+        results = data.read()
+        data.close()
+
+        results = json.loads(results)
+        return results
+
+    def _getResultZipped(self, uri, callname="", clearCache=True, args=None):
+        """
+        This method is retrieving a zipped file from the uri privided, instead
+        of the normal json
+        :param uri: The endpoint uri
+        :return:    A dictionary
+        """
```

Comment:
How about defining a `raise NotImplemented` or something like that? Please check the documentation, I do not remember whether that is the correct exception name.

Reply:
That's a good idea! doing it now.